brainpad/courses/vb/variables_and_nested_statements.md

```diff
@@ -1 +1,426 @@
+* Supplies: BrainPad and microUSB cable.
+
+## Arguments
+Arguments are values passed on to methods. For example, to set the frequency of the buzzer, a value is passed on to that method. The 3000 is the arguments.
+
+```
+BrainPad.Buzzer.Start(3000)
+```
+Arguments are also used when adding delays. Here, the 0.2 is the argument.
+```
+BrainPad.Wait.Seconds(0.2)
+```
+Variables can be constants, like the numbers used above. They can also be variables, which will be explained in the coming sections. Also, arguments can also be something completely different!
+
+```
+    Public Sub BrainPadSetup()
+        'Declared but not used
+    End Sub
+
+    Public Sub BrainPadLoop()
+        If BrainPad.Buttons.IsPressed(BrainPad.Buttons.Button.Up) Then
+            BrainPad.WriteDebugMessage("Up button is pressed")
+        Else
+            BrainPad.WriteDebugMessage("Up button is not pressed")
+        End If
+    End Sub
+```
+What is that argument used with `Button.IsPressed()`? In this case, the `Button.IsPressed` takes an argument to specify which button is to be checked.
+
+This code 
+```
+BrainPad.Buttons.IsPressed(BrainPad.Buttons.Button.Up)
+```
+Is much clearer than:
+```
+BrainPad.Button.IsPressed(4)
+```
+What is button 4? In fact, internally the up button maybe number 4 but remembering this will be difficult.
+
+## Exercise
+Try to use the wrong argument type then build the project to observe the error messages.
+
+## Integer Variables
+Variable is a location in memory that is reserved for us to use. Instead of knowing where the memory is and how it is located, the system gives us an access to that memory using a name we select.
+```
+Dim count As Integer
+```
+The above line instructs the system to reserve (Dim) a variable that is named count and the variable type is an Integer. The name can be use any letters from the alphabet and any numbers but it must not start with a number and it can�t contain spaces.
+
+Here are few allowed name examples:
+```
+Dim count123 As Integer
+Dim countMeIn As Integer
+Dim count2var As Integer
+```
+Variables in small programs can use simply names. It can simply be just `x`. However, using a meaningful name is very important when writing any serious program, as there maybe tens, or hundreds, of variables.
+
+The variable now can store numbers.
+Note that the variable can be used in the equation and in the result as well.
+```
+count = count + 2
+```
+
+The above code will add 2 to the count.
+
+Math can be used as well.
+```
+count = 5 + result
+```
+
+
+After selecting an appropriate name, an appropriate type is needed. The type tell the system how much memory to reserve for that one variable. The most widely used variable is an Integer, and can hold very large numbers, even negative numbers. We suggest the use for Integers for everything, except when fractions are needed. More on fractions later. 
+
+## Changing Tones
+Use the code below:
+
+```
```

Comment:
We should show the class declaration here, especially since we're declaring variables in the class.